### ISSUE TITLE
Streamline User, UserFriends, and FriendTag relationships

### ISSUE THREAD

**Issue by SeabertYuan** [Daggerpov/Spawn-App-Back-End#63]
1. Delete UserFriendTagMapping and UserFriends
2. Refactor and consolidate the above into FriendTag and UserFriendTag
3. Restructure db to only have user, friend_tag, user_friend_tags tables

Reason is on slack, but TL:DR, there's a lot of overlap and unnecessary models which add complexity and don't match how the app frontend is designed.

Edit:
UserFriendTag is actually what I intended UserFriendTagMapping to be, so I changed those names.

**Comment by Daggerpov** [comment]
From @SeabertYuan on Slack:

I think we can get away with three tables: user, friend_tags, user_friend_tag_mapping.

**TLDR**:
1. Delete UserFriendTag and UserFriends
2. Refactor and consolidate the above into FriendTag
3. Restructure db to only have user, friend_tag, user_friend_tag_mapping tables

**Why**?
The idea is we keep the user table the same, and we make a friend_tags table which stores the friend tag details (color, name, id, etc.) but crucially stores the “owner” (the User’s id who owns the tag). Then for the user_friend_tag_mapping table, we can store a column of FriendTag id’s and a column of associated User ids.

If we remove the user_friends table, we can still access this information by storing UserFriends as an “everyone” FriendTag in the user_friend_tag_mapping, and friend_tags table.

This way we have less going on in our relationship diagrams.

**Comment by Daggerpov** [comment]
Done in #50
